Fixed issue Multi-Tree Deleting Nodes outside the Scope

jonmc1 opened the pull request:
GitHub Issue #11289

jonmc1 committed:
Fixed issue Multi-Tree Deleting Nodes outside the Scope

GitHub Issue #11289

markstory commented on src/ORM/Behavior/TreeBehavior.php:
```diff
@@ -224,12 +224,16 @@ public function beforeDelete(Event $event, EntityInterface $entity)
+                })
+                ->clause('where');
+
+            $this->_table->deleteAll($expression);
```
Instead of picking off the where clause you might want to use a subquery. The default find method _could_ add conditions from associations which will fail with this approach.

markstory replied:
I pushed up a commit with the changes I was proposing and a test case for this issue.

markstory committed:
Use a subquery to delete child records.

Picking off the where clause is risky as the default find() method may
contain associations or joins.

ADmad reviewed (changes requested):
Tests failing

markstory commented:
Oh MySQL. You so silly. I'll get that fixed up.

markstory committed:
Don't use a subquery for deletes.

MySQL doesn't like deleting from a table that it is also selecting
records from. Use lower level query APIs to get the desired results.
